### PR TITLE
US808 - As a developer, I want to refactor the Movement Manager to removed unused strategy abstraction & stabalize the movement execution

### PR DESCRIPTION
**Pull request by nicholasnastase**
#809Delete IMovementStrategy and components that reference it, remove interface inheritance from StandardMovementStrategy
#811Replace all references to IMovementStrategy with direct StandardMovementStrategy references
#812Update documentation for StandardMovementStrategy to reflect interface removal and revised responsibilities
#813Refactor direct callers of movement logic to use concrete movement components
#814Clean up StandardMovementStrategy that is no longer being used (using statements, comments, anything abstraction/interface related
#815Review StandardMovementStrategy methods to keep only the direct movement entry points needed by callers
#816Validate movement tests still run after refactor

### CODE REVIEW

**Review by hdathert — CHANGES_REQUESTED** [review]
This feels really incomplete- there are so many commented out event channels all over the file, and it is still doing a *lot* more than it probably should be. Its not clear the scope of its responsibility- is this meant to be a public API for `PlayerController`s to manage movement? If so, then why is it tracking dice rolls?

I would go back to the audit on this one and reevaluate what needs to be done here.

File: Assets/Scripts/Managers/Movement/StandardMovementStrategy.cs
Comment (hdathert) [inline Assets/Scripts/Managers/Movement/StandardMovementStrategy.cs]: How is this method meant to be called in the normal turn order? Is this meant to be called by the player controller? Right now it looks like it was subscribed to an event channel that is no longer in use, so this is a dangling method.

If its meant to be called by the `PlayerController`s, then that's an issue since right now, `PlayerController`s have *no idea* what this component is and have no way of accessing it.

This is the kind of thing that was the target of the audit- figuring out what this is, and how its meant to be used, and then aligning the component to that task and purpose.

Possible solutions is to make this a static method (like other utilities), which would involve a substantial refactor of the class, but since it involves state (doubles rolled), we'd have to change how we count doubles to make that work... 

Alternatively, we could make it a singleton with a static access-er to ensure that only one exists but enables calls from `PlayerController` without needing direct references, turning it into a service effectively. If we wanted to do this, we'd probably want to refactor the entry method's name and we'd have to ensure that player controller is 'blocked' until movement is fully completed.

Another solution is to keep it as a component and include a reference to it inside `PlayerController`, which already has a pretty substantial constructor.

Or we could make use of event channels like we were before, and have this called indirectly via events from the `PlayerController`, but we'd need some sort of call back to the `PlayerController` so it knows when movement is done, and if they're allowed to roll again.

All of these options have pros and cons in terms of complexity, but I think the Singleton method would be the easiest to implement in terms of decoupling (no event channels, low infrastructure needs), but would have to deal with async method completion, since `PlayerController` and the Movement needs to wait for the movement animations to complete.

**Review by hdathert — CHANGES_REQUESTED** [review]
This still needs more work- its not clear how this class is meant to be used and currently does not at all function in the architecture.

File: Assets/Scripts/Managers/Movement/StandardMovementStrategy.cs [hunk 1]
```diff
@@ -1,218 +1,267 @@
...
             landed?.OnLanded(currentPlayer);
 
-            //handles the doubles roll logic
             if (doublesCount > 0)
             {
-                Logger.Debug("StandardMovementStrategy", "Doubles → allow re-roll", LogCategory.Gameplay, this);
-                //this is where we can put a re-roll button...
-                //recall also, we defined 3 doubles rolls is a jail sentence!
+                Logger.Debug("StandardMovementStrategy.ResolveCompletedMovement",
+                    "Doubles rolled. Extra roll remains available to current flow.",
```
Comment (hdathert) [inline Assets/Scripts/Managers/Movement/StandardMovementStrategy.cs]: technically, doubles means they get an extra turn in our system- so its not really the same thing. This is more semantics on the log messages though.

**Review by hdathert — APPROVED** [review]
Approved. If you can, definitely fix the log line so its semantically correct to the system, and also fix the merge conflicts that are stemming from the major `GameManager` overhauls.

**Review by hdathert — APPROVED** [review]

**Comment by tsaldrid** [comment]
# Code Review Checklist

**Completed for**: US #809 and Tasks #811 to #816  
**Code Review Completed By**: Tsaldrid

Use this checklist to review new code changes before merging.

---

## Implementation
- [x] Does this code change accomplish what it is supposed to do?
- [x] Can the solution be simplified?
- [x] This change does **not** add unwanted compile-time or run-time dependencies.
- [x] The code is at the right abstraction level.
- [x] The code is modular enough.
- [x] The solution is adequate for maintainability, readability, performance, and security.
- [x] Similar functionality does **not** already exist in the codebase (or reuse is not appropriate).
- [x] No additional best practices or design patterns would substantially improve this code.

---

## Logic Errors and Bugs
- [x] No use cases where the code behaves incorrectly.
- [x] No inputs or external events that can break the code.
- [x] Code agrees with specifications.
- [x] No off-by-one errors.
- [x] No magic numbers (or they are well-named constants).
- [x] Proper use of global/large-scope variables (minimized and justified).

---

## Error Handling and Logging
- [x] Error handling is implemented correctly.
- [x] Error messages are user-friendly (where applicable).
- [x] Sufficient, meaningful log events exist for debugging (no sensitive data in logs).

---

## Dependencies
- [x] Documentation/configuration/README updates were made as needed.
- [x] No negative impact on other parts of the system or backward compatibility.

---

## Security and Data Privacy
- [x] No new security vulnerabilities introduced.
- [x] All user input is validated, sanitized, and escaped (XSS/SQL injection prevented).
- [x] Data from external APIs/libraries is handled safely and checked as needed.

---

## Performance
- [x] No significant performance regressions.
- [x] No obvious opportunities for significant performance gains left unaddressed.

---

## Ethics and Morality
- [x] Appropriate measures exist to prevent or report harassment/abuse (if applicable).

---

## Testing and Testability
- [x] Code is testable.
- [x] Automated tests have been added or updated.
- [x] Existing tests sufficiently cover the change (unit/integration/system).
- [x] Edge cases and boundary inputs are tested.

---

## Readability
- [x] Code is easy to understand.
- [x] Methods/functions are appropriately small and focused.
- [x] Names (functions/classes/variables) are clear and descriptive.
- [x] Code is in the correct file/folder/package.
- [x] Control flow is intuitive; data flow is clear.
- [x] Comments are clear, relevant, and up to date.
- [x] No commented-out or redundant code remains.
- [x] No overly dense one-liners; no overly long methods.

---

**Use this before submitting a pull request.**  
It helps ensure code quality, maintainability, and compliance with best practices.

*Based on awesome code review guides and MIT’s code review recommendations.*
